src/Interpreters/HashJoin.cpp

```diff
@@ -1020,12 +1018,12 @@ class AddedColumns
         {
             for (size_t j = 0, size = right_indexes.size(); j < size; ++j)
             {
-                auto column_from_block = block.getByPosition(right_indexes[j]);
```

Comment:
This line of code is the regression root cause. We should not do any heavy operation (block copy) inside hot loop.

Reply:
Thanks for fixing and explaining the issue!
I didn't know about the `insertFromFullColumn` but it's really similar to what we do with `nullable` columns so I should've expected for something like that to exist.